Revise StatefulSet tutorial (#42767)
kubernetes/website: content/en/docs/tutorials/stateful-application/basic-stateful-set.md

## Creating a StatefulSet

> Begin by creating a StatefulSet (and the Service that it relies upon) using

Reviewer: Perhaps?
```suggestion
Begin by creating a StatefulSet (and a Service to expose it) using
```

### Ordered Pod creation

> A StatefulSet defaults to creating its Pods in a strict order.

Reviewer: ```suggestion
By default, a StatefulSet will create Pods in a strict order.
```

The Pods' ordinals, hostnames, SRV records, and A record names have not changed,
but the IP addresses associated with the Pods may have changed. In the cluster
used for this tutorial, they have. This is why it is important not to configure
other applications to connect to Pods in a StatefulSet by the IP address

> of a particular Pod (it is OK to connect to Pods by resolving their hostname).

Reviewer: ```suggestion
of a particular Pod. It is OK to instead connect to Pods by resolving their hostname.
```

application will be able to discover the Pods' addresses when they transition
to Running and Ready.

If your application wants to find any healthy Pod in a StatefulSet,

> and therefore does not need to track each specific Pod,

Reviewer 2: Cannot understand what this is saying ...

Author: OK, the intent is to describe applications that want to connect to something like a database made up of stateful Pods, but the client wants an endpoint representing the database cluster. The client in this example is not interested in tracking each potential backend endpoint and maintaining its own view about which of those backends is healthy.

The Kubernetes _answer_ is load balancing (using kube-proxy, or using round-robin DNS), to avoid the client-side load balancing that a client would otherwise need to do.

> you could also connect to the IP address of a `type: ClusterIP` backed

Reviewer 2: incomplete sentence ... are you saying that a client can connect to the (Service) backed
by ... ?

Author: This sentence looks complete to me.

by the Pods in that StatefulSet. You can use the same Service that
tracks the StatefulSet (specified in the `serviceName` of the StatefulSet)

> or, if your use case needs it, a separate Service that selects the right
> set of Pods.

Reviewer 2: we don't have to be so wordy ...

```suggestion
or a separate Service that selects the right set of Pods.
```

application will be able to discover the Pods' addresses when they transition
to Running and Ready.

If your application wants to find any healthy Pod in a StatefulSet,
and therefore does not need to track each specific Pod,

> you could also connect to the IP address of a `type: ClusterIP` Service,
> backed by the Pods in that StatefulSet. You can use the same Service that

Reviewer: ```suggestion
you could also connect to the IP address of a `type: ClusterIP` Service
which exposes the Pods in that StatefulSet. You can use the same Service that
```

## Scaling a StatefulSet

Scaling a StatefulSet refers to increasing or decreasing the number of replicas
(horizontal scaling).
This is accomplished by updating the `replicas` field. You can use either
[`kubectl scale`](/docs/reference/generated/kubectl/kubectl-commands/#scale) or
[`kubectl patch`](/docs/reference/generated/kubectl/kubectl-commands/#patch) to scale a StatefulSet.

### Scaling up

> Scaling up means adding more replicas. If you balance work across the

Reviewer 2: I would disagree to this definition.
My understanding is that horizontal scaling is about scaling in or scaling out;
while vertical scaling is about scaling up or scaling down.

> StatefulSet, the new larger set of Pods can perform more of it.

Reviewer 2: "If you .... of it".
This sentence is not clarifying any thing.
It in fact introduce some confusion to users, i.e. how am I supposed to balance work across a SatefuleSet? Why do we call that larger set a new set?
Yes, the intent to add more replicas is crystal clear, isn't it?

Author: (I tried that new text)

the status of a rolling update to a StatefulSet
{{< /note >}}

#### Staging an update

You can split updates to a StatefulSet that uses the `RollingUpdate` strategy
into _partitions_, by specifying `.spec.updateStrategy.rollingUpdate.partition`.

For more context, you can read [Partitioned rolling updates](/docs/concepts/workloads/controllers/statefulset/#partitions)
in the StatefulSet concept page.

You can stage an update to a StatefulSet by using the `partition` field within
`.spec.updateStrategy.rollingUpdate`.
For this update, you will keep the existing Pods in the StatefulSet using

> unchanged whilst you change the pod template for the StatefulSet.

Reviewer 2: "using unchanged" ???

Author: I'll mark this resolved. Let me know if that doesn't sound right.

kubectl get pod -l app=nginx --watch
```

In another terminal, reconfigure the StatefulSet for `Parallel` Pod management:

```shell
kubectl apply -f https://k8s.io/examples/application/web/web-parallel.yaml
```
```
service/nginx updated
statefulset.apps/web updated
```

Keep the terminal open where you're running the watch. In another terminal window, scale the
StatefulSet:

```shell
kubectl scale statefulset/web --replicas=5
```
```
statefulset.apps/web scaled
```

> Examine the output of the terminal where the `kubectl get` command is running. It may look something likeL

Reviewer 3: nit
```suggestion
Examine the output of the terminal where the `kubectl get` command is running. It may look something like
```
